### PR TITLE
Documentation: Annotation skeletons

### PR DESCRIPTION
**Pull request by mdacoca**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/_index.md [hunk 1]
```diff
@@ -2,15 +2,23 @@
...
+possessing unique attributes and properties like color, occlusion, and visibility.
+
+However, a **Skeleton** point can only exist within its parent **Skeleton**.
+
+**Skeleton** elements can be marked as `outside` and hidden if needed,
+such as when a part is outside the frame.
+
+**Skeletons** can be [**exported**](/docs/manual/advanced/formats/)
+in two formats: [**CVAT for image**](/docs/manual/advanced/formats/format-cvat/#cvat-for-videos-export)
+and [**COCO Keypoints**](/docs/manual/advanced/formats/coco-keypoints/).
```
[the remark this answers is not in the document]
Comment (mdacoca) [reply site/content/en/docs/manual/advanced/annotation-with-skeletons/_index.md:24]: Thank you! will update

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/_index.md [hunk 2]
```diff
@@ -2,15 +2,23 @@
...
-Any skeleton elements can be hidden (by marking them `outside`) if necessary (for example if a part is out of a frame).
-Currently there are two formats which support exporting skeletons: CVAT & COCO.
+A **Skeleton** is composed of numerous points (also referred to as elements),
+which may be connected by edges. Each point functions as an individual object,
+possessing unique attributes and properties like color, occlusion, and visibility.
+
+However, a **Skeleton** point can only exist within its parent **Skeleton**.
+
+**Skeleton** elements can be marked as `outside` and hidden if needed,
+such as when a part is outside the frame.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/_index.md:20]: `outside` is also skeleton property like occlusion or color. It is not totally clear why it mentioned after one paragraph. 

Also there is some mess between visibility, occluded and outside. Maybe to clarify that in details. 
Both outside and visibility may do a skeleton point invisible. But outside property is part of annotations, when visibility makes a point hidden only for annotator's convenience and not saved somehow between different sessions.

Occluded and outside are similar in their meaning, but `occluded` keeps point visible on the frame and usually means that point still on a frame, just hidden by some upper object. Outside is recommended to be used when part of object is out of frame borders.

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+In this guide, we delve into the efficient process of annotating complex
+structures through the implementation of **Skeleton** automatic annotations.
+
+See:
+
+- [Creating auto annotation Skeleton task](#creating-auto-annotation-skeleton-task)
+- [Automatic annotation with Skeletons](#automatic-annotation-with-skeletons)
+  - [Editing skeletons on the canvas](#editing-skeletons-on-the-canvas)
+
+## Creating auto annotation Skeleton task
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:17]: I believe it should be described somewhere in labels constructor documentation.
It is just one more alternative way to add a label (any label, skeleton is only particular case) to the task. And it is not really related to skeletons automatic annotations

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   you want to use.
+4. Click on the model's labels, you want to use.
+   Selected labels will become gray.
+
+   ![](/images/auto-annot-sk.jpg)
+
+5. Click **Done**. Labels, that you selected,
+   will appear in the labels window.
+6. (Optional) If you want to adjust labels, within the
+   label, click the **Update** attributes icon.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:33]: P.S. Skeleton points currently may be adjusted only before a task/project created.

Comment (bsekachev) [reply site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:33]: It means that if to pick a skeleton label from a model in created task/project, user will not be able to adjust them.

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md (hunk 3 above)
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:17]: What is ``auto annotation Skeleton task``? How does it differ from ``manual annotation Skeleton task``?

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+8. Click:
+   - **Submit & Open** to create and open the task.
+   - **Submit & Continue** to submit the configuration and start creating a new task.
+
+## Automatic annotation with Skeletons
+
+To automatically annotate with **Skeleton**, do the following
+
+1. Open job and on the tools panel select **AI Tools** > **Detectors**
+2. From the drop-down list select model.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:46]: Probably we need to mention model name here. In app.cvat.ai it is ``Human pose estimation`` model. 

And, I believe before this we need to show some steps on how to deploy this model.

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+   You will see list of points to match and
+   the name of the label on the top of the list.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:48]:
```suggestion
   You will see list of points to match and
   the name of the skeleton on the top of the list.
```

Comment (bsekachev) [reply site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:48]: That is not actually true. 

1. User will see mapping configurator to map model skeletons to task skeletons. For each mapped skeleton user will see mapping configurator for points of paired skeletons.
2. But those skeletons that have the same name in the model, and in the task, will be mapped automatically. In points mapping configurator, points from paired skeletons with the same name, also will be mapped automatically.

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+   ![](/images/auto-annot-sk-detectors.jpg)
+
+3. (Optional) Remove point that you
+   do not need, by clicking on the
+   bin icon.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:54]: I would describe it in general: 
User can remove any mapped item (a skeleton together with all points, or certain points from two mapped skeletons)

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+4. Click **Annotate**.
+
+### Editing skeletons on the canvas
+
+Individual skeleton points can be adjusted independently.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/auto-skeletons.md:59]: This line looks useless, I believe link to general skeleton configuration is enough here.

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+5. Use **Add point** to add points to the **Drawing area**.
+6. Use **Add edge** to add edges between points.
+7. Upload files.
+8. Click:
+   - **Submit & Open** to create and open the task.
+   - **Submit & Continue** to submit the configuration and start creating a new task.
+
+### Configuring Skeleton points
+
+You can add labels, attributes and change colour of any point of the skeleton.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md:73]:
```suggestion
You can name label, setup attributes, and change colour of each point of the skeleton.
```

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+  This action is applicable for shortcuts like lock, occluded,
+  pinned, keyframe, and outside for skeleton tracks.
+- To use a shortcut to a specific skeleton point, hover over the
+  point and push the shortcut keyboard key.
+  The same list of shortcuts is available, with the addition of outside,
+  which is also applicable to individual skeleton shape elements.
+
+### Editing skeletons on the sidebar
+
+Шn CVAT, the sidebar offers an alternative method for setting up skeleton properties and attributes.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md:131]:
```suggestion
In CVAT, the sidebar offers an alternative method for setting up skeleton properties and attributes.
```

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+> You can copy/insert skeletons configuration from the **Raw** tab of the label configurator.
+
+## Annotation with Skeletons
+
+To annotate with **Skeleton**, do the following
+
+1. Open job.
+2. On the tools panel select **Draw new skeleton**.
+3. Select **Track** to track object or **Shape** to annotate
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md:94]: Such words make me think that CVAT will do tracking the object for me automatically

File: site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+| **3**  | **Click and drag**          | Use it to move points across the **Drawing area** (**8**).                                                                            |
+| **4**  | **Add edge**                | Use it to add edge on the **Drawing area** (**8**) to connect the points (**2**).                                                     |
+| **5**  | **Remove point**            | Use it to remove points. Click on **Remove point** and then on any point (**2**) on the **Drawing area** (**8**) to delete the point. |
+| **6**  | **Download skeleton**       | Use it to download created skeleton in .SVG format.                                                                                   |
+| **7**  | **Upload skeleton**         | Use it to upload skeleton in .SVG format.                                                                                             |
+| **7**  | **Drawing area**            | Use it as a canvas to draw a skeleton.                                                                                                |
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/annotation-with-skeletons/manual-skeletons.md:50]:
```suggestion
| **8**  | **Drawing area**            | Use it as a canvas to draw a skeleton.                                                                                                |
```

File: site/content/en/docs/manual/advanced/skeletons.md [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+- [Editing skeletons on the sidebar](#editing-skeletons-on-the-sidebar)
+
+## Skeleton task
+
+To start annotating using skeletons, you need to set up a **Skeleton** task
+in **Configurator**:
+
+To open **Configurator**, when [creating a task](/docs/manual/basics/create_an_annotation_task/),
+click on the **Setup skeleton** button if you want to set up the skeleton manually,
+or **From model** if you want to add skeleton labels from the model.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/skeletons.md:44]:
```suggestion
or **From model** if you want to add skeleton labels from a model.
```

File: site/content/en/docs/manual/advanced/skeletons.md [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+| **5**  | **Remove point**            | Use it to remove points. Click on **Remove point** and then on any point (**2**) on the **Drawing area** (**8**) to delete the point. |
+| **6**  | **Download skeleton**       | Use it to download created skeleton in .SVG format.                                                                                   |
+| **7**  | **Upload skeleton**         | Use it to upload skeleton in .SVG format.                                                                                             |
+| **8**  | **Drawing area**            | Use it as a canvas to draw a skeleton.                                                                                                |
+
+<!--lint enable maximum-line-length-->
+
+You can add **Skeleton** points manually or use the model.
+
+### Adding Skeleton manually
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/skeletons.md:69]: I feel like Skeleton configurator should be described exactly in this section, not above

File: site/content/en/docs/manual/advanced/skeletons.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+   - **Submit & Continue** to submit the configuration and start creating a new task.
+
+### Adding Skeleton labels from the model
+
+To add points from the model, and annotate do the following:
+
+1. Open **Basic configurator**.
+2. On the **Constructor** tab, click **From model**.
+3. From the **Select a model to pick labels** select the
+   `Human pose estimation` model.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/skeletons.md:93]:
```suggestion
   `Human pose estimation` model or others if available.
```

File: site/content/en/docs/manual/advanced/skeletons.md [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+4. Click on the model's labels, you want to use.
+   <br>Selected labels will become gray.
+
+   ![](/images/auto-annot-sk.jpg)
+
+5. (Optional) If you want to adjust labels, within the
+   label, click the **Update** attributes icon.
+   <br>The [**Skeleton configurator**](/docs/manual/advanced/skeletons/#skeleton-task)
+   will open, where you can [**configure the skeleton**](/docs/manual/advanced/skeletons/#configuring-skeleton-points).
+   <br>**Note**: Labels cannot be adjusted after the task is created.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/skeletons.md:103]:
```suggestion
   <br>**Note**: Labels cannot be adjusted after the task/project is created.
```

File: site/content/en/docs/manual/advanced/skeletons.md [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+6. Click **Done**. The labels, that you selected,
+   will appear in the labels window. 6. Upload data.
+7. Click:
+   - **Submit & Open** to create and open the task.
+   - **Submit & Continue** to submit the configuration and start creating a new task.
```
Comment (bsekachev) [inline site/content/en/docs/manual/advanced/skeletons.md:108]:
```suggestion
   will appear in the labels window. 
7. Upload data.
8. Click:
   - **Submit & Open** to create and open the task.
   - **Submit & Continue** to submit the configuration and start creating a new task.
```

**Review by bsekachev — APPROVED** [review]

**Pull request merged**